### PR TITLE
Functional tests - Crud contact

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add crud contact tests (Add, update and delete)
| Type?         | refacto
| Category?     |TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/shopParameters/contact/02_CRUDContact.js" URL_FO=SHOPURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16962)

### CODE REVIEW

File: tests/puppeteer/pages/BO/shopParameters/contact/add.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  /**
+   * Fill form for add/edit contact
+   * @param contactData
+   * @return {Promise<textContent>}
+   */
+  async createEditContact(contactData) {
+    await this.setValue(this.titleInput, contactData.firstName);
+    await this.setValue(this.emailAddressInput, contactData.email);
+    await this.setValue(this.descriptionTextarea, contactData.description);
+    if (contactData.saveMessage) await this.page.click(this.enableSaveMessageslabel.replace('%ID', '1'));
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/contact/add.js]:
```suggestion
    await this.page.click(this.enableSaveMessageslabel.replace('%ID', saveMessage ? 1 : 0));
```

File: tests/puppeteer/pages/BO/shopParameters/contact/add.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    // Save Contact
+    await this.clickAndWaitForNavigation(this.saveContactButton);
+    await this.page.waitForSelector(this.alertSuccessBlockParagraph, {visible: true});
+    return this.getTextContent(this.alertSuccessBlockParagraph);
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/contact/add.js]: No need to wait for this selector `alertSuccessBlockParagraph`

File: tests/puppeteer/pages/BO/shopParameters/contact/add.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  /*
+  Methods
+   */
+
+  /**
+   * Fill form for add/edit contact
+   * @param contactData
+   * @return {Promise<textContent>}
+   */
+  async createEditContact(contactData) {
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/contact/add.js:48]: can you fill form in both languages

File: tests/puppeteer/pages/BO/shopParameters/contact/index.js [hunk 4]
```diff
@@ -18,6 +20,11 @@ module.exports = class Contacts extends BOBasePage {
     this.contactFilterInput = `${this.contactsListForm} #contact_%FILTERBY`;
     this.filterSearchButton = `${this.contactsListForm} button[name='contact[actions][search]']`;
     this.filterResetButton = `${this.contactsListForm} button[name='contact[actions][reset]']`;
+    // Actions buttons in Row
+    this.listTableToggleDropDown = `${this.contactsListTableColumn.replace('%COLUMN', 'actions')}
+    a[data-toggle='dropdown']`;
+    this.listTableEditLink = `${this.contactsListTableColumn.replace('%COLUMN', 'actions')} a[href*='edit']`;
+    this.deleteRowLink = `${this.contactsListTableColumn.replace('%COLUMN', 'actions')} a[data-method='POST']`;
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/contact/index.js]:
```js
    this.contactsListTableActionsColumn = this.contactsListTableColumn.replace('%COLUMN', 'actions');
    this.listTableToggleDropDown = `${this.contactsListTableActionsColumn} a[data-toggle='dropdown']`;
    this.listTableEditLink = `${this.contactsListTableActionsColumn} a[href*='edit']`;
    this.deleteRowLink = `${this.contactsListTableActionsColumn} a[data-method='POST']`;
```

File: tests/puppeteer/pages/BO/shopParameters/contact/index.js [hunk 5]
```diff
@@ -76,4 +83,50 @@ module.exports = class Contacts extends BOBasePage {
...
+   * @return {Promise<void>}
+   */
+  async goToEditContactPage(row) {
+    // Click on dropDown
+    await Promise.all([
+      this.page.click(this.listTableToggleDropDown.replace('%ROW', row)),
+      this.page.waitForSelector(`${this.listTableToggleDropDown
+        .replace('%ROW', row)}[aria-expanded='true']`, {visible: true},
+      ),
+    ]);
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/contact/index.js]:
```js
    await Promise.all([
      this.page.click(this.listTableToggleDropDown.replace('%ROW', row)),
      this.page.waitForSelector(
        `${this.listTableToggleDropDown.replace('%ROW', row)}[aria-expanded='true']`,
        {visible: true},
      ),
    ]);
```

File: tests/puppeteer/pages/BO/shopParameters/contact/index.js [hunk 6]
```diff
@@ -76,4 +83,50 @@ module.exports = class Contacts extends BOBasePage {
...
+    this.dialogListener();
+    // Click on dropDown
+    await Promise.all([
+      this.page.click(this.listTableToggleDropDown.replace('%ROW', row)),
+      this.page.waitForSelector(
+        `${this.listTableToggleDropDown
+          .replace('%ROW', row)}[aria-expanded='true']`,
+        {visible: true},
+      ),
+    ]);
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/contact/index.js]:
```js
    await Promise.all([
      this.page.click(this.listTableToggleDropDown.replace('%ROW', row)),
      this.page.waitForSelector(
        `${this.listTableToggleDropDown.replace('%ROW', row)}[aria-expanded='true']`,
        {visible: true},
      ),
    ]);
```

**Review by matthieu-rolland — APPROVED** [review]
lgtm

File: tests/puppeteer/pages/BO/shopParameters/contact/add.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+      this.page.waitForSelector(`${this.pageTitleLangButton}[aria-expanded='false']`, {visible: true}),
+    ]);
+  }
+
+  /**
+   * Fill form for add/edit contact
+   * @param contactData
+   * @return {Promise<textContent>}
+   */
+  async createEditContact(contactData) {
```
Comment (Progi1984) [inline tests/puppeteer/pages/BO/shopParameters/contact/add.js]: Why not `createContact` ?

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**